packages/colorpicker/lib/src/colorpicker.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:colorpicker/src/constants/colors.dart';
+import 'package:colorpicker/src/components/color_compare.dart';
+import 'package:colorpicker/src/components/slider.dart';
+import 'package:component_library/component_library.dart';
+import 'package:flutter/material.dart';
+
+class ColorPicker extends StatefulWidget {
+  const ColorPicker({
+    super.key,
+    required this.currentColor,
+    required this.onColorChanged,
+  });
+
+  final Color currentColor;
+  final void Function(Color) onColorChanged;
+
+  @override
+  State<ColorPicker> createState() => _ColorPickerState();
+}
+
+class _ColorPickerState extends State<ColorPicker> {
+  final colors = DisplayColors.colors;
+  Color newColor = Colors.black;
+  double opacity = 1.0;
+
+  void callback(Color color, double op) {
+    setState(() {
+      opacity = op;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    newColor = widget.currentColor;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: const EdgeInsets.all(26.0),
+      alignment: Alignment.center,
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(16.0),
+      ),
+      child: Column(
+        children: [
+          ColorCompare(
+            currentColor: widget.currentColor,
+            newColor: newColor.withOpacity(opacity),
+          ),
+          const SizedBox(
+            height: 10.0,
+          ),
+          GridView.count(
+            childAspectRatio: 1.4,
+            crossAxisCount: 4,
+            crossAxisSpacing: 2.0,
+            mainAxisSpacing: 2.0,
+            shrinkWrap: true,
+            children: List.generate(
+              colors.length + 1,
+              (index) {
+                if (index == colors.length) {
+                  return Container(
+                    decoration: BoxDecoration(
+                      image: DecorationImage(
+                        image: CheckerboardImg.getCheckerboardImgAsset(),
+                        fit: BoxFit.contain,
+                        repeat: ImageRepeat.repeat,
+                      ),
+                    ),
+                  );
+                }
+                return GestureDetector(
+                  onTap: () {
+                    setState(() {
+                      newColor = colors[index];
+                    });
+                  },
```

Comment:
Should we use riverpod here? @bakicelebi

Reply:
okay so should I use riverpod in the whole package?

Reply:
Yes, we are using riverpod for our state management. Also try to use generated riverpod providers.  https://riverpod.dev/de/docs/concepts/about_code_generation

We will be refactoring the non-generated providers in the future. You can look at brush tool state provider as an example. 

Cheers. 🦇

Reply:
okay sure, will do it asap.. thanks!